From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dmitry Sinyavin <sinyavin@gmail.com>
Date: Sat, 25 Jul 2026 00:00:00 +0000
Subject: [PATCH] dt-bindings: mtd: qcom,nandc: add MDM9607

MDM9607 contains a QPIC v1 NAND controller using BAM DMA.  Unlike the
other integrations, the firmware device tree exposes only the QPIC core
clock; there is no separate AON clock input.

Signed-off-by: Dmitry Sinyavin <sinyavin@gmail.com>
---
 .../devicetree/bindings/mtd/qcom,nandc.yaml   | 26 ++++++++++++++++--
 1 file changed, 24 insertions(+), 2 deletions(-)

--- a/Documentation/devicetree/bindings/mtd/qcom,nandc.yaml
+++ b/Documentation/devicetree/bindings/mtd/qcom,nandc.yaml
@@ -23,16 +23,19 @@ properties:
               - qcom,ipq6018-nand
               - qcom,ipq8074-nand
               - qcom,sdx55-nand
+              - qcom,mdm9607-nand
 
   reg:
     maxItems: 1
 
   clocks:
+    minItems: 1
     items:
       - description: Core Clock
       - description: Always ON Clock
 
   clock-names:
+    minItems: 1
     items:
       - const: core
       - const: aon
@@ -87,6 +90,31 @@ allOf:
   - $ref: nand-controller.yaml#
 
   - if:
+      not:
+        properties:
+          compatible:
+            contains:
+              const: qcom,mdm9607-nand
+    then:
+      properties:
+        clocks:
+          minItems: 2
+        clock-names:
+          minItems: 2
+
+  - if:
+      properties:
+        compatible:
+          contains:
+            const: qcom,mdm9607-nand
+    then:
+      properties:
+        clocks:
+          maxItems: 1
+        clock-names:
+          maxItems: 1
+
+  - if:
       properties:
         compatible:
           contains:
@@ -122,6 +150,7 @@ allOf:
               - qcom,ipq6018-nand
               - qcom,ipq8074-nand
               - qcom,sdx55-nand
+              - qcom,mdm9607-nand
 
     then:
       properties:
